src/nupic/encoders/ScalarSensor.cpp

```diff
@@ -94,7 +94,12 @@ namespace nupic
   void ScalarSensor::compute()
   {
     Real32* array = (Real32*)encodedOutput_->getData().getBuffer();
-    const Int32 iBucket = encoder_->encodeIntoArray(sensedValue_, array);
+    UInt uintArray[encoder_->getOutputWidth()];
+    for(UInt i=0; i<encoder_->getOutputWidth(); i++) //FIXME optimize
+    {
+      uintArray[i] = (UInt)array[i];
+    }
+    const Int32 iBucket = encoder_->encodeIntoArray(sensedValue_, uintArray);
```

Comment:
For the moment, I don't support this change. But I'll also point out that this code doesn't work. You're never copying the encoding to the output `array`. So the Spatial Pooler isn't receiving any input.